Std/Data/List/Basic.lean

```diff
@@ -521,6 +533,29 @@ Split a list at every occurrence of a separator element. The separators are not
 -/
 @[inline] def splitOn [BEq α] (a : α) (as : List α) : List (List α) := as.splitOnP (· == a)
 
+/--
+Split a list at every occurrence of a separator sublist. The separators are not in the result.
+```
+[1, 1, 2, 3, 2, 4, 4].splitOnSublist [2, 3] = [[1, 1], [2, 4, 4]]
+```
+-/
+def splitOnSublist [BEq α] (l sep : List α) : List (List α) := go l sep sep [] where
```

Comment:
This name is confusing because we currently use `sublist` to mean a non-contiguous subsequence, see `List.Sublist`. How about just `splitOnList`?

Reply:
Sounds great. I'll rename it after I finish other work.